diff --git a/plugins/artsearch/artsearch.py b/plugins/artsearch/artsearch.py
index 5e5e3e7..d704795 100644
--- a/plugins/artsearch/artsearch.py
+++ b/plugins/artsearch/artsearch.py
@@ -24,6 +24,11 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301  USA.
 
+import gi
+
+gi.require_version('Peas', '1.0')
+gi.require_version('RB', '3.0')
+
 from gi.repository import GObject, Peas, RB, GdkPixbuf
 
 import gettext
diff --git a/plugins/rb/rb.py b/plugins/rb/rb.py
index e956dd4..4f2ae16 100644
--- a/plugins/rb/rb.py
+++ b/plugins/rb/rb.py
@@ -31,6 +31,10 @@ import os.path
 import os
 import time
 
+import gi
+
+gi.require_version('RB', '3.0')
+
 from gi.repository import RB, Gtk
 
 # rb classes
